staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -283,6 +286,12 @@ func (flags *ApplyFlags) ToOptions(f cmdutil.Factory, cmd *cobra.Command, baseNa
 	}
 
 	openAPISchema, _ := f.OpenAPISchema()
+	var openAPIV3Root openapi3.Root
+	openAPIV3Client, err := f.OpenAPIV3Client()
+	if err == nil {
+		cachedOpenAPIV3Client := cachedopenapi.NewClient(openAPIV3Client)
+		openAPIV3Root = openapi3.NewRoot(cachedOpenAPIV3Client)
```

Comment:
Do we need to get the root here?

Reply:
NewRoot creates the interface for fetching the OpenAPI V3 documents. It doesn't make a call to the apiserver. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/openapi3/root.go#L69

Reply:
Thanks, that's great